backend/app/models/experiment_template.py

```diff
@@ -36,17 +42,72 @@ class ExperimentTemplate(Document):
     retry_count: int = 0
     approved: bool = False
     created_by: str
+    is_usable: bool = True
```

Comment:
I don't get this flag much... I see there is a dedicated endpoint that can change this flag and nothing else. User will be flagging the template? We need to discuss the logic here and perhaps change the naming to something clear.

Reply:
Sure thing, the main idea (for ExperimentTemplates at least) is that once there exist experiments tied to this template, you cant really physically delete it from DB. We could potentionally could rename `is_usable` to `is_deleted` and flip its meaning, but I was hesitant to actually call it that way as there are actually some templates that could be physically removed from DB. 

Usually you don't want to delete anything from DB, but I decided to allow this if there are no experiments tied to that specific template. In this case we can get rid of templates that have never and will never be used.

One more thing, even the unusable templates (`is_usable = False`) can still be seen in `My templates` section, however the ones you physically delete cannot be seen, since you know, they were removed from the DB itself.